Jenkinsfile

```diff
@@ -41,19 +42,38 @@ pipeline {
     IROHA_POSTGRES_PORT = 5432
   }
 
+  triggers {
+        parameterizedCron('''
+0 23 * * * %BUILD_TYPE=Release; Linux=True; MacOS=True; ARMv7=False; ARMv8=True; Nightly=True; Doxygen=False; JavaBindings=False; PythonBindings=False; BindingsOnly=False; PARALLELISM=4
+        ''')
+    }
   options {
     buildDiscarder(logRotator(numToKeepStr: '20'))
   }
 
   agent any
   stages {
-    stage ('Stop same job builds') {
+    stage ('Stop bad job builds') {
       agent { label 'master' }
       steps {
         script {
-          // Stop same job running builds if any
-          def builds = load ".jenkinsci/cancel-builds-same-job.groovy"
-          builds.cancelSameJobBuilds()
+          if (BRANCH_NAME != "develop") {
+            if (params.Nightly) {
+                // Stop this job running if it is nightly but not the develop it should be
+                def tmp = load ".jenkinsci/cancel-nightly-except-develop.groovy"
+                tmp.cancelThisJob()
+            }
+            else {
+              // Stop same job running builds if it is commit/PR build and not triggered as nightly
+              def builds = load ".jenkinsci/cancel-builds-same-job.groovy"
+              builds.cancelSameJobBuilds()
```

Comment:
Please, do not cancel builds if they're already running. Let commit continue building and run nightly build alongside already running one.